### PR TITLE
Add PriceFormatt and Test

### PR DESCRIPTION
**Pull request by MateoRicoCalvo**
I refactored the way we handle and display event prices across the application to ensure consistency, precision, and proper currency formatting.

1. Problem

Previously, the event price rendering logic was duplicated in multiple components such as EventCard and EventDetails, and prices were interpreted directly as integers representing full Swiss francs. This approach had two main issues:

Code duplication: each component implemented its own price rendering logic.

Lack of precision: it was not possible to represent or format partial francs (e.g., CHF 12.50) correctly.

2. Solution

To fix this, I introduced a unified formatting utility responsible for converting and displaying prices consistently throughout the app.

 I created a new utility class. Object PriceFormatter

Prices are now interpreted as UInt values representing cents (1/100 of a franc).

This avoids floating-point arithmetic and ensures exact currency representation.

The logic also gracefully handles free events (displaying "Free" when the price is zero).

3. Integration

I updated both EventCard and EventDetails components to use the new formatter.    This ensures that all prices are rendered using the same centralized logic, improving maintainability and reducing the chance of formatting inconsistencies.

4. Testing

To validate the behavior of the new utility, I added a comprehensive unit test suite (PriceFormatterTest) covering multiple edge cases:

Zero price (“Free”)

Integer franc values (e.g., 300u ->CHF 3.00)

Partial francs (305u -> CHF 3.05, 1250u -> CHF 12.50)

Small fractional prices (75u  ->CHF 0.75)

Large amounts (123456u -> CHF 1234.56)

All tests passed successfully, confirming the formatter behaves as expected across different input values.

### CODE REVIEW

File: app/src/main/java/ch/epfllife/model/event/Event.kt [hunk 1]
```diff
@@ -10,6 +10,6 @@ data class Event(
     val time: String, // for the moment it will be a string
     val associationId: String,
     val tags: Set<String>,
-    val price: UInt = 0u,
+    val price: UInt = 0u,   // price in cents
```
Comment (AlecGhost) [inline app/src/main/java/ch/epfllife/model/event/Event.kt]: Instead of attaching this information in a comment, I think it would make sense to actually store the value in a new type `Price` which abstracts over the representation. This class can then have the formatting method, instead of this price formatter object.

File: app/src/main/java/ch/epfllife/ui/composables/Price.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+class Price {
+
+  val price: UInt
+
+  constructor(price: UInt) {
+    this.price = price
+  }
+
+  @SuppressLint("DefaultLocale")
+  fun formatPrice(): String {
```
Comment (GustavOnsberg) [inline app/src/main/java/ch/epfllife/ui/composables/Price.kt]: You can override the toString function instead
